### PR TITLE
[riverpod_graph] Support generated family providers

### PR DESCRIPTION
**Pull request by ValentinVignal**
Adds the support for generated family providers and family class providers

image

### CODE REVIEW

**Comment by ValentinVignal** [comment]
@rrousselGit Is there something I should do about this PR?

File: packages/riverpod_graph/lib/src/analyze.dart [hunk 1]
```diff
@@ -38,9 +39,19 @@ Future<void> analyze(String rootDirectory) async {
...
+          if (variableElement.type.element?.isFromRiverpod ?? false) {
+            return true;
+          }
+          return variableElement.type.extendsFromRiverpod;
+          //        ||
+          // (this is ClassElement &&
+          //     (this as ClassElement)
+          //         .thisType
+          //         .allSupertypes
+          //         .any((superType) => superType.element.isFromRiverpod))
```
Comment (rrousselGit) [inline packages/riverpod_graph/lib/src/analyze.dart:52]:
```suggestion
```

**Comment by rrousselGit** [comment]
Nope. Just me taking a bit of time to review

**Comment by rrousselGit** [comment]
Thanks for your work!

**Pull request merged**